sourcery_analytics/extractors.py

```diff
@@ -159,6 +167,29 @@ def _extract_from_directory(self, directory: pathlib.Path) -> typing.Iterator[T]
         )
 
     def _extract_from_file(self, file: pathlib.Path) -> typing.Iterator[T]:
-        manager = astroid.manager.AstroidManager()
-        module = manager.ast_from_file(file)
-        yield from self._extract_from_node(module)
+        try:
+            module = self.manager.ast_from_file(file)
+            yield from self._extract_from_node(module)
+        except astroid.AstroidSyntaxError as e:
+            if isinstance(e.error, SyntaxError):
+                error_message = _format_syntax_error_message(
+                    "skipping file", file, e.error
+                )
+            else:
+                error_message = str(e).replace("\n", " ")
+            warning = SyntaxWarning(error_message)
+            warnings.warn(warning)
+            yield from ()
```

Comment:
**suggestion (edge_case_not_handled):** Ensure proper handling of syntax errors for robustness.

Catching syntax errors and warning the user is a good practice. However, consider if there are any additional steps that should be taken to ensure the robustness of the application. For example, should the extraction attempt to continue with other files, or should it halt? Documenting the intended behavior in a comment here would be helpful.

```suggestion
        except astroid.AstroidSyntaxError as e:
            if isinstance(e.error, SyntaxError):
                error_message = _format_syntax_error_message(
                    "skipping file due to syntax error", file, e.error
                )
            else:
                error_message = str(e).replace("\n", " ")
            warning = SyntaxWarning(error_message)
            warnings.warn(warning)
            # Documenting behavior: When a syntax error is encountered, a warning is issued,
            # and the extraction process skips the current file but continues with others.
            # This ensures the application remains robust by not halting the entire extraction
            # process due to errors in individual files.
            yield from ()
```